ingest/indexer/service/blockprocessor/full_indexer_block_process_strategy.go

```diff
@@ -25,6 +27,17 @@ func (f *fullIndexerBlockProcessStrategy) IsFullBlockProcessor() bool {
 
 // ProcessBlock implements commondomain.BlockProcessStrategy.
 func (f *fullIndexerBlockProcessStrategy) ProcessBlock(ctx sdk.Context) (err error) {
+	// If block processor is a full block processor, check if the node is syncing
+	// If node is syncing, skip processing the block (which publishes token supplies and pools data)
+	// We can wait until node is synced to publish the token supplies and pools data as we only need the latest snapshot of it
+	isNodeSyncing, err := f.nodeStatusChecker.IsNodeSyncing(ctx)
+	if err != nil {
+		return &commondomain.NodeSyncCheckError{Err: err}
+	}
+	if isNodeSyncing {
+		return commondomain.ErrNodeIsSyncing
+	}
+
```

Comment:
Notes: this is one of the core logic in this PR, it checks whether sync is still in progress and if so, return `ErrNodeIsSyncing` error to the caller